client/src/components/cart/CartTable.js

```diff
@@ -0,0 +1,187 @@
+import { useContext } from "react";
+import styled from "styled-components";
+import Payment from "../forms/Payment";
+import { Link } from "react-router-dom";
+import { CartContext } from "../context/Context";
+function CartTable() {
+  const [cart, setCart] = useContext(CartContext);
+  const toggleItemQuantity = (itemName, subtract) => {
+    let newState;
+    if (subtract) {
+      newState = cart.map((item) => {
+        item.name === itemName && item.quantity--;
+        return item;
+      });
+    } else {
+      newState = cart.map((item) => {
+        item.name === itemName && item.quantity++;
+        return item;
+      });
+    }
+    setCart(newState);
+  };
+  const handleDelete = (name) => {
+    const filteredCart = cart.filter((item) => item.name != name);
+    setCart(filteredCart);
+  };
+  return (
+    <TableContainer>
+      <h2>Your Cart</h2>
+      {cart.length ? (
+        <>
+          <table>
+            <thead>
+              <tr>
+                <th>Product</th>
+                <th>Price</th>
+                <th>Quantity</th>
+                <th>Total</th>
+              </tr>
+            </thead>
+            <tbody>
+              {cart.map((item, index) => (
+                <tr key={index}>
+                  <td className="flex">
+                    <img src={item.imageUrl} alt="" />
+                    <h4>{item.name}</h4>
+                  </td>
+                  <td>{item.price}</td>
+                  <td>
+                    <span
+                      onClick={() =>
+                        item.quantity > 1 &&
+                        toggleItemQuantity(item.name, "subtract", item.quantity)
+                      }
+                      className="minus"
+                    >
+                      -
```

Comment:
Another option is to not show the `-` button or to make it unclickable (should be obvious from the UI that it isn't clickable too, like grey it out a bit), when there are no items in the cart. 

Based on the current implementation, there's no UI indication that clicking it shouldn't work, so if I click and nothing happens, I may wonder why for a minute before I realise I have zero items in my cart.

Reply:
Currently, the cart doesn't even get to zero, because I just go tired of the logic. Ideally, if it gets to zero, the item should be deleted from the cart I believe. Will implement it, Thanks!